packages/web_benchmarks/lib/client.dart

```diff
@@ -278,7 +279,7 @@ class LocalBenchmarkServerClient {
     // 404 is expected in the following cases:
     // - The benchmark is ran using plain `flutter run`, which does not provide "next-benchmark" handler.
     // - We ran all benchmarks and the benchmark is telling us there are no more benchmarks to run.
-    if (request.status == 404) {
+    if (request.responseText == kEndOfBenchmarks || request.status == 404) {
```

Comment:
This works, but it seems that "benchmarks ended" and "there is no server" are considered together as a single case. Are there any downsides to this?

Reply:
If we wanted a different outcome, we could separate them, but for now since both lead to the manual fallback there's no distinction. I wouldn't have the 404 case if I didn't want to support running benchmarks using `flutter run`. It's the only reason it's there.